File: v1/src/main/java/com/api/v1/customers/domain/CustomerCrudRepository.java
package com.api.v1.customers.domain;

import java.util.Optional;

import com.api.v1.customers.domain.exposed.Customer;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.mongodb.repository.Query;

public interface CustomerCrudRepository extends MongoRepository<Customer, String> {

    @Query("{ 'person.sin': ?0 }")
    Optional<Customer> findBySIN(String sin);

    @Query("{ 'person.email': ?0 }")
    Optional<Customer> findByEmail(String email);
}

File: v2/src/main/kotlin/com/api/v2/customers/domain/CustomerCrudRepository.kt
package com.api.v2.customers.domain

import com.api.v2.customers.Customer
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.mongodb.repository.Query

interface CustomerCrudRepository: MongoRepository<Customer, String> {

    @Query("{ 'person.sin': ?0 }")
    fun findBySIN(sin: String): Customer?

    @Query("{ 'person.email': ?0 }")
    fun findByEmail(email: String): Customer?
}